src/main/java/at/hannibal2/skyhanni/features/misc/skillprogress/SkillProgress.kt

```diff
@@ -0,0 +1,138 @@
+import at.hannibal2.skyhanni.utils.NumberUtil.roundToPrecision
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStringsAndItems
+import at.hannibal2.skyhanni.utils.SpecialColour
+import at.hannibal2.skyhanni.utils.StringUtils.firstLetterUppercase
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import io.github.moulberry.notenoughupdates.util.Utils
+import net.minecraft.init.Items
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+import kotlin.math.ceil
+import kotlin.time.Duration.Companion.seconds
+
+class SkillProgress {
+
+    private val config get() = SkyHanniMod.feature.misc.skillProgressDisplayConfig
+    private var skillExpPercentage = 0.0
+    private var display = emptyList<List<Any>>()
+    private val defaultStack = Utils.createItemStack(Items.banner, "Default")
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (!showDisplay) return
+        config.position.renderStringsAndItems(display, itemScale = 1.5, posLabel = "Skill Progress")
+        if (config.showProgressBar.get() && display.isNotEmpty()) {
+            val progress = if (config.useTexturedBar.get()) {
+                var factor = skillExpPercentage.toFloat()
+                if (factor > 1f) factor = 1f
+                factor *= 182f
+                Renderable.texturedProgressBar(factor, Color(SpecialColour.specialToChromaRGB(config.barStartColor)), width = 182, height = 5, useChroma = config.useChroma.get())
+            } else
+                Renderable.progressBar(skillExpPercentage, Color(SpecialColour.specialToChromaRGB(config.barStartColor)), Color(SpecialColour.specialToChromaRGB(config.barStartColor)), width = 182)
+
+            config.barPosition.renderRenderables(listOf(progress), posLabel = "Skill Progress Bar")
+        }
+    }
+
+    @SubscribeEvent
+    fun onProfileSwitch(event: PreProfileSwitchEvent) {
+        display = emptyList()
+        skillExpPercentage = 0.0
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled()) return
+
+        if (lastUpdate.passedSince() > 3.seconds) showDisplay = config.alwaysShow.get()
+
+        if (event.repeatSeconds(1)) {
+            update()
+        }
+    }
+
+    private fun update() {
+        display = drawDisplay()
+    }
+
+    private fun drawDisplay(): List<List<Any>> {
+        val newDisplay = mutableListOf<List<Any>>()
+        val skillMap = skillMap ?: return newDisplay
+        val skill = skillMap[activeSkill] ?: return newDisplay
+
+        newDisplay.add(buildList {
+            if (config.showLevel.get())
+                add("§9[§d${skill.level}§9] ")
+
+            if (config.useIcon.get())
+                add(stackMap.getOrDefault(activeSkill.firstLetterUppercase(), defaultStack))
```

Comment:
Apply the itemStack scale directly not implicit with the render.
```suggestion
                add(Renderable.itemStack(stackMap.getOrDefault(activeSkill.firstLetterUppercase(), defaultStack)),1.5)
```